executor/extension/validator/transaction_validator.go

```diff
@@ -133,9 +134,13 @@ func (v *stateDbValidator) runPreTxValidation(tool string, db state.VmStateDB, s
 		return nil
 	}
 
-	err := validateWorldState(v.cfg, db, state.Data.GetInputState(), v.log)
-	if err == nil {
-		return nil
+	var err error
+	if strings.ToLower(v.cfg.UpdateOnFailure) == "all" || strings.ToLower(v.cfg.UpdateOnFailure) == "pre" {
```

Comment:
calling `ToLower` two times every tx is a bit expensive. I would call `ToLower` while assigning to the config itself, here you will be 100% sure its lowercase...